src/main/java/wolox/training/controllers/BookController.java

```diff
@@ -1,16 +1,90 @@
 package wolox.training.controllers;
 
-import org.springframework.stereotype.Controller;
+import java.util.List;
+import javax.validation.Valid;
+import org.postgresql.util.PSQLException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
 import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.DeleteMapping;
 import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import wolox.training.constants.ExceptionsConstants;
+import wolox.training.exceptions.BookNotFoundException;
+import wolox.training.exceptions.DatabaseException;
+import wolox.training.exceptions.IdMismatchException;
+import wolox.training.models.Book;
+import wolox.training.repositories.BookRepository;
 
-@Controller
+@RestController
+@RequestMapping("api/books")
 public class BookController {
 
+    @Autowired
+    private BookRepository bookRepository;
+
     @GetMapping("/greeting")
     public String greeting(@RequestParam(name="name", required=false, defaultValue="World") String name, Model model) {
         model.addAttribute("name", name);
         return "greeting";
     }
+
+    @GetMapping
+    public ResponseEntity<List<Book>> getAll() {
+        return new ResponseEntity<>(bookRepository.findAll(), HttpStatus.OK);
+    }
+
+    @GetMapping("{id}")
+    public ResponseEntity<Book> get(@PathVariable int id) throws BookNotFoundException{
+        return new ResponseEntity<>(bookRepository.findById(id).orElseThrow(() -> new BookNotFoundException(
+                ExceptionsConstants.BOOK_NOT_FOUND)), HttpStatus.OK);
+    }
+
+    @PostMapping
+    public ResponseEntity<Book> create(@Valid @RequestBody Book book) throws DatabaseException {
+        try {
+            return new ResponseEntity<>(bookRepository.save(book), HttpStatus.CREATED);
+        } catch (Exception e) {
+
+            throw new DatabaseException(ExceptionsConstants.DATA_SAVE_INTEGRITY_VIOLATION);
+        }
+
+    }
+
+    @PutMapping("{id}")
+    public ResponseEntity<Book> update(@PathVariable int id,@Valid @RequestBody Book book) throws BookNotFoundException, IdMismatchException, DatabaseException {
+        if(!bookRepository.existsById(id)) {
```

Comment:
I would throw the IdMismatchException first, so if this happens, you will avoid getting to the database to check if there's a book with the provided id, which will be useless at the end. (Principle of [fail fast](https://en.wikipedia.org/wiki/Fail-fast) 🤓 )
Also, by making this change, if the IdMismatchException is not thrown, you can be sure that you are actually looking for the right id on the DB. I mean, as it is right now, when you check existence, you cannot be sure that `id` and `book.id` are equal, so why are you only checking the existence using the param `id` and not by `book.id`. (I posted my trail of thoughts just so you could understand my suggestion, it's not wrong but you did, but you could improve it 💪 )